lib/src/main/java/dev/openfeature/javasdk/OpenFeatureClient.java

```diff
@@ -59,8 +59,12 @@ <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key, T defa
             this.beforeHooks(hookCtx, mergedHooks, hints);
 
             ProviderEvaluation<T> providerEval;
+            // I suppose eventually all the types will go here? looks like only boolean is handled so far, or am I missing something huge?
+            // If so, it would be nice to find some way of avoiding a big switch case, but that may not be reasonable in Java
             if (type == FlagValueType.BOOLEAN) {
                 // TODO: Can we guarantee that defaultValue is a boolean? If not, how to we handle that?
+                // the compiler will guarantee this no? Since the Features interface ensures that the default matches the return value?
```

Comment:
I've not had a chance to really think through it. I *think* it's fine, but I'm only like 70% confident.